### DISCUSSION TITLE
Exodus output within specific time intervals (Control System or...?)

### DISCUSSION THREAD

**Discussion by gka80** [idaholab/moose#23221] [Q&A General] [answered]
Hello-

I have an instance where it would be beneficial to output exodus files only within specific time intervals. The main reason why I want to do this is to save storage space.

I am wondering if this is possible and, if so, how? I have tried with the Control System using "BoolFunctionControl", but I am not sure how to tie this with the "Outputs/exodus = true" input.

```
[./exodus_bool]
    type = ParsedFunction
    value = 'if(t>=0.0,1,0)*if(t<0.1,1,0)'
  [../]

[Controls]
  [./exodus_control]
    type = BoolFunctionControl
    parameter = 'Ouput/*/exodus'
    function = 'exodus_bool'
    execute_on = 'initial timestep_begin'
  [../]
[]

[Outputs]
  # exodus = true
  [exodus]
    type = Exodus
    exodus = true
  []
[]
```

However, when I run this, I get the following error:

```
*** ERROR ***
The following error occurred in the object "exodus_control", of type "BoolFunctionControl".

The desired parameter 'Ouput/*/exodus' was not located for the 'exodus_control' object, it either does not exist or has not been declared as controllable.
```

Can I control output this way? Or is there another (preferred) approach?

Thank you,

-Garrett

**Comment by GiudGiud** [comment, marked as answer]
Hello

This is more of a conditional enable control, not a boolean function control. 

You want to turn on/off the output (exodus) object. You can use this object
https://mooseframework.inl.gov/source/controls/ConditionalFunctionEnableControl.html

BUT
you dont need to do that
If you look at the parameters for the exodus output
https://mooseframework.inl.gov/source/outputs/Exodus.html
there are already a start_time and end_time parameter.

Guillaume

**Reply by gka80** [reply]
@GiudGiud 

Thank you! Both options worked great. 

For future reference if anyone needs to do this using the `ConditionalFunctionEnableControl`:

```
[Functions]
  [./exodus_control]
      type = ParsedFunction
      value = '((t >= 0.0)&(t < 0.00128))|((t >= 0.00256)&(t < 0.00576))'
  [../]
[]

[Controls]
  [./exodus_control]
    type = ConditionalFunctionEnableControl
    conditional_function = 'exodus_control'
    enable_objects = 'Outputs::exodus'
  [../]
[]

[Outputs]
  [./exodus]
    type = Exodus
    enable = true
  [../]
[]
```
